src/solver/variable/state.cpp

```diff
@@ -353,117 +352,149 @@ void State::yearEndBuildFromThermalClusterIndex(const uint clusterAreaWideIndex,
     }
 
     if (dur > 0)
+        yearEndBuildFromThermalClusterIndexDurPositive(dur, ON_min, ON_max, ON_opt);
+
+    // Calculation of non linear and startup costs
+    yearEndBuildThermalClusterCalculateStartupCosts(dur, ON_min, ON_opt, currentCluster);
+}
+
+void State::yearEndBuildThermalClusterCalculateStartupCosts(uint dur,
```

Comment:
we should pass **std::array**s by reference to **const** : 
`const std::array<uint, Variable::maxHoursInAYear>& ON_min`